tests/libvmi/lifecycle.go

```diff
@@ -0,0 +1,46 @@
+package libvmi
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+)
+
+func SetupVMI(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+	virtClient, err := kubecli.GetKubevirtClient()
+	Expect(err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
+
+	By("Creating the VMI")
+	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+
+	By("Waiting until the VMI gets ready")
+	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
+
+	return vmi
+}
+
+func CleanupVMI(vmi *v1.VirtualMachineInstance) {
+	if vmi != nil {
```

Comment:
what about adding a defer func that would nil the vmi in case it wasnt nil ?

Reply:
that it would behave better in case the ptr was already deleted in previous round but didnt get a new value in the current round.
however it wont work if the ptr is passed by value.

Reply:
You answered when it would "behave better" but I asked how is it better, what would be the benefit of changing it. Be specific. Give me an example and explain what is broken and how would you improvement solve it.

Reply:
i answered already "better in case the ptr was already deleted in previous round but didnt get a new value in the current round."

Since each test run it as part of AfterEach,  leftovers of the previous test can be reused, in case current test didnt give it a new value. unless we nil it.

If test1 ran - created a vmi and deleted it
and then test2 from the same `Context` ran, failed on `Setup` function and didnt give a value to the ptr, (for example if someone added an `Expect` there or if there was a panic there, or if someone will add logic that can fail before the `Setup`)
the old value of the ptr would be used in order to determine if we need to clean again.

the improvement should we to nil the Ptr after the call to Clean or as the first line on BeforeEach (there are tests that do this exactly).

its up to you. you can just assume it wont happen / that we dont care for exceptions or wrong logic on the AfterEach as other PRs decided.